Partition_2/include/CGAL/Partition_traits_2.h

```diff
@@ -38,58 +38,26 @@ class Partition_traits_2  : public Partition_traits_2_base<Kernel_>
   private:
     typedef Kernel_                                     Kernel;
     typedef Partition_traits_2<Kernel_>                 Self;
+  
   public:
     typedef typename Kernel::Point_2                    Point_2;
     typedef ::std::list<Point_2>                        Container;
     typedef CGAL::Polygon_2<Kernel, Container>          Polygon_2;
     typedef typename Kernel::Less_yx_2                  Less_yx_2;
     typedef typename Kernel::Less_xy_2                  Less_xy_2;
-    typedef typename Kernel::Left_turn_2                 Left_turn_2;
+    typedef typename Kernel::Left_turn_2                Left_turn_2;
     typedef typename Kernel::Orientation_2              Orientation_2;
     typedef typename Kernel::Compare_y_2                Compare_y_2;
     typedef typename Kernel::Compare_x_2                Compare_x_2;
     typedef CGAL::Is_convex_2<Self>                     Is_convex_2;
     typedef CGAL::Is_y_monotone_2<Self>                 Is_y_monotone_2;
 
-    // needed by Indirect_edge_compare, used in y_monotone and greene_approx
-    typedef typename Kernel::Line_2                     Line_2;
-    typedef typename Kernel::Construct_line_2           Construct_line_2;
-    typedef typename Kernel::Compare_x_at_y_2           Compare_x_at_y_2;
-    typedef typename Kernel::Is_horizontal_2            Is_horizontal_2;
 
     // needed by visibility graph and thus by optimal convex
-    typedef typename Kernel::Ray_2                      Ray_2; 
     typedef typename Kernel::Collinear_are_ordered_along_line_2
                                             Collinear_are_ordered_along_line_2;
     typedef typename Kernel::Are_strictly_ordered_along_line_2
                                             Are_strictly_ordered_along_line_2;
```

Comment:
Not a consequence of the PR, but the concept really doesn't describe the actual requirements of the algorithm. There's zero chance the code compiles with a fake traits meeting the bare requirements of the concepts.